src/storage/statistics/column_statistics.cpp

```diff
@@ -20,8 +20,7 @@ shared_ptr<ColumnStatistics> ColumnStatistics::CreateEmptyStats(const LogicalTyp
 
 void ColumnStatistics::Merge(ColumnStatistics &other) {
 	stats.Merge(other.stats);
-	if (distinct_stats) {
-		D_ASSERT(other.distinct_stats);
+	if (distinct_stats && other.distinct_stats) {
```

Comment:
If other is null, is it possible there are still DistinctStats but something in the logic to create them returned NULL? If so, it seems possible that not updating distinct stats with these stats could cause the join order optimizer to produce bad plans since the stats do not include the distinct information from `other`.

I imagine this is a rare case though, since otherwise this bug would have been reported earlier and have an easy reproducer